sync2/handler2/handler.go

```diff
@@ -150,12 +154,11 @@ func (h *Handler) OnTerminated(userID, deviceID string) {
 	h.updateMetrics()
 }
 
-func (h *Handler) OnExpiredToken(userID, deviceID string) {
-	h.v2Store.RemoveDevice(deviceID)
-	h.Store.ToDeviceTable.DeleteAllMessagesForDevice(deviceID)
-	h.Store.DeviceDataTable.DeleteDevice(userID, deviceID)
-	// also notify v3 side so it can remove the connection from ConnMap
+func (h *Handler) OnExpiredToken(accessTokenHash, userID, deviceID string) {
+	h.v2Store.TokensTable.Delete(accessTokenHash)
+	// Notify v3 side so it can remove the connection from ConnMap
 	h.v2Pub.Notify(pubsub.ChanV2, &pubsub.V2ExpiredToken{
+		UserID:   userID,
 		DeviceID: deviceID,
 	})
```

Comment:
At first I didn't delete the token, but then the token would get selected on incoming requests only for the homeserver to reject it. Loop of doom.

Do we care about deleting messages etc? We could have some kind of cleanup that nukes messages for devices if they've had no access tokens for at least e.g. 30 days.